docs/api.yml

```diff
@@ -0,0 +1,414 @@
+openapi: 3.0.0
+
+info:
+  title: OSM Teams
+  description: Teams for OpenStreetMap!
+  version: 0.0.0
+
+servers:
+  - url: https://dev.mapping.team/api
+    description: Beta version of the API
+
+consumes:
+  - application/json
+
+produces:
+  - application/json
+
+tags:
+  - name: teams
+    description: Teams
+  - name: clients
+    description: Clients
+
+components:
+  schemas:
+    Team:
+      properties:
+        id:
+          type: integer
+          format: int64
+          description: unique id of the team
+        name:
+          type: string
+          description: name of the team
+        hashtag:
+          type: string
+          description: hashtag representing the team
+        bio:
+          type: string
+          description: description of the team
+        privacy:
+          type: string
+          enum: [private, public]
+          description: if `private`, team details/members are not shown in ui unless user is a member of the team
+        require_join_request:
+          type: boolean
+          description: if true, this team requires potential new members to request access
+        created_at:
+          type: string
+          format: date-time
+          description: the timestamp of the time and date the team was created
+        updated_at:
+          type: string
+          format: date-time
+          description: the timestamp of the time and date the team was last updated
+        location:
+          type: string
+          description: geojson point
+    TeamList:
+      type: array
+      items:
+        $ref: '#/components/schemas/Team'
+    TeamMember:
+      properties:
+        id:
+          type: string
+          description: OSM id of user
+        name:
+          type: string
+          description: OSM username
+    TeamMemberList:
+      type: array
+      items:
+        $ref: '#/components/schemas/TeamMember'
+    TeamModerator:
+      properties:
+        id:
+          type: integer
+          description: unique id representing the team moderator relation
+        osm_id:
+          type: integer
+          description: OSM id of user
+        team_id:
+          type: integer
+          description: unique id of the team
+    TeamModeratorList:
+      type: array
+      items:
+        $ref: '#/components/schemas/TeamModerator'
+    TeamFullDetail:
```

Comment:
Open to suggestions on the name of this schema. The difference is that when requesting an individual team through `/teams/:id` we get a team object with members and moderators, but when getting a list of teams, the team objects do not include members and moderators. So `TeamFullDetail` includes members & moderators while `Team` does not.